javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt

```diff
@@ -0,0 +1,83 @@
+package arcs.core.allocator
+
+import arcs.core.common.Id
+import arcs.core.data.CreatableStorageKey
+import arcs.core.data.Plan
+import arcs.core.host.MultiplePersonPlan
+import arcs.core.host.PersonPlan
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import arcs.core.storage.driver.RamDisk
+import com.google.common.truth.Truth
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class DefaultStorageKeyCreatorTest {
+
+  @Before
+  open fun setUp() = runBlocking {
+    RamDisk.clear()
+    DriverAndKeyConfigurator.configure(null)
+  }
+
+  // Simple Case.
+  @Test
+  fun keyCreator_createStorageKeysIfNecessary() = creatorKeyTest(PersonPlan)
+
+  // Complex case.
+  @Test
+  fun keyCreator_createStorageKeysIfNecessary_manyParticles() = creatorKeyTest(MultiplePersonPlan)
+
+  /**
+   * Given a [Plan], test that it's keys start as [CreatableStorageKey]s and are transformed
+   * using the [DefaultStorageKeyCreator.createStorageKeysIfNecessary] method. Verify that the plan
+   * returned
+   */
+  private fun creatorKeyTest(plan: Plan) {
+    val idGenerator = Id.Generator.newSession()
+    val testArcId = idGenerator.newArcId("Test")
+    val keyCreator = DefaultStorageKeyCreator()
+
+    // Verify the handle storage keys start as creatable.
```

Comment:
The tests above shouldn't be called "IfNecessary" if we're enforcing that all storage keys start as creatable; I'm guessing the name comes from the method name under test? In which case, we should be testing examples where not every key needs to be created too (I acknowledge we're sort of doing this here by calling createStorageKeysIfNecessary twice, but a plan with a mixture of creatable and non-creatable keys isn't currently tested).

Reply:
Added a mixed test as well.

Reply:
you should actually be able to create plans that have pre-created storage keys (e.g. if they reference storage from other plans).